src/super_context.ml

```diff
@@ -160,31 +170,34 @@ let create
     |> List.filter ~f:(fun s -> not (String.is_prefix s ~prefix:"-std="))
   in
   let vars =
+    let path_exp p cos = Action.Var_expansion.Paths ([p], cos) in
+    let str_exp s cos = Action.Var_expansion.Strings (s, cos) in
     let ocamlopt =
       match context.ocamlopt with
       | None -> Path.relative context.ocaml_bin "ocamlopt"
       | Some p -> p
     in
     let make =
       match Bin.make with
-      | None   -> "make"
-      | Some p -> Path.to_string p
+      | None   -> path_exp (Path.of_string "make")
+      | Some p -> path_exp p
     in
-    [ "-verbose"       , "" (*"-verbose";*)
-    ; "CPP"            , sprintf "%s %s -E" context.c_compiler context.ocamlc_cflags
-    ; "PA_CPP"         , sprintf "%s %s -undef -traditional -x c -E" context.c_compiler
-                           context.ocamlc_cflags
-    ; "CC"             , sprintf "%s %s" context.c_compiler context.ocamlc_cflags
-    ; "CXX"            , String.concat ~sep:" " (context.c_compiler :: cxx_flags)
-    ; "ocaml_bin"      , Path.to_string context.ocaml_bin
-    ; "OCAML"          , Path.to_string context.ocaml
-    ; "OCAMLC"         , Path.to_string context.ocamlc
-    ; "OCAMLOPT"       , Path.to_string ocamlopt
-    ; "ocaml_version"  , context.version
-    ; "ocaml_where"    , Path.to_string context.stdlib_dir
-    ; "ARCH_SIXTYFOUR" , string_of_bool context.arch_sixtyfour
+    let cflags = String.extract_blank_separated_words context.ocamlc_cflags in
+    [ "-verbose"       , str_exp [] (*"-verbose";*)
+    ; "CPP"            , str_exp (context.c_compiler :: cflags @ ["-E"])
+    ; "PA_CPP"         , str_exp (context.c_compiler :: cflags
+                                  @ ["-undef -traditional"; "-x"; "c"; "-E"])
```

Comment:
Shouldn't `"-undef -traditional"` be `"-undef"; "-traditional"`?

Reply:
Yes.